File: LEETCODE/1. Two Sum/Solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i,x in enumerate(nums):
            z = target - x
            if z in nums[i+1:]:
                for j in range(i+1,len(nums)):
                    if nums[j]==z:
                        break
                break
        return [i,j]
        
        
        z1 = nums.index(n[i])
        z2 = nums.index(n[j])
        return [z1,z2]
            
        

File: LEETCODE/1. Two Sum/Solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> map = new HashMap<>();
        
        for(int i=0;i<nums.length;i++){
            Integer idx = map.get(nums[i]);
            
            if(idx==null)
                map.put(target-nums[i],i);
            else
                return new int[]{idx,i};
        }
        
        return new int[2];
    }
}